crates/rpc/rpc/src/eth/api/fees.rs

```diff
@@ -185,4 +187,92 @@ where
             reward: Some(rewards),
         })
     }
+
+    /// Populate the fee history cache when new blocks are added to the chain
+    ///
+    /// This listens for any new blocks and updates the fee history cache
+    async fn populate_fee_history_cache_on_new_block<St>(
+        &mut self,
+        reward_percentiles: Option<Vec<f64>>,
+        mut events: St,
+    ) where
+        St: Stream<Item = CanonStateNotification> + Unpin + 'static,
```

Comment:
this is almost the right interface

I'd like to have this as separate function that also takes ownership of `self`